x/ccv/provider/keeper/relay.go

```diff
@@ -55,7 +58,13 @@ func (k Keeper) OnTimeoutPacket(ctx sdk.Context, packet channeltypes.Packet) err
 
 // EndBlockVSU contains the EndBlock logic needed for
 // the Validator Set Update sub-protocol
-func (k Keeper) EndBlockVSU(ctx sdk.Context) {
+func (k Keeper) EndBlockVSU(ctx sdk.Context) ([]abci.ValidatorUpdate, error) {
+	// logic to update the provider consensus validator set.
+	// Important: must be called before the rest of EndBlockVSU, because
+	// we need to know the updated provider validator set
```

Comment:
Not sure I fully understand this comment. In `QueueVSCPacket`s we would need to compute the minimum power in the Top N. However, there we use directly `GetLastActiveBondedValidators` that talks directly to the `staking` keeper and we do not seem to be using the stored provider consensus validator set for this. Why would we need to update the provider consensus validator set to make sure that the minimum power is computed correctly?

Reply:
You are right, I changed that and didn't adjust the comment. Have removed the comment